WC_Mds_Shipping_Method.php

```diff
@@ -194,47 +194,48 @@ public function calculate_shipping($package = [])
                     $services = $this->collivery->getServices();
                     if (is_array($services)) {
                         // Get pricing for each service
-                        foreach ($services as $id => $title) {
-                            if ($this->mdsSettings->getInstanceValue("method_$id") == 'yes') {
+                        foreach ($services as $service) {//$id => $title) {
+                            if ($this->mdsSettings->getInstanceValue("method_".$service['id']) == 'yes') {
                                 // Now lets get the price for
-                                $riskCover = 0;
+                                $riskCover = false;
                                 $adjustedTotal = $package['shipping_cart_total'];
                                 $riskCoverEnabled = $this->mdsSettings->getValue( 'risk_cover' ) == 'yes';
                                 $overThreshold = $adjustedTotal >= $this->mdsSettings->getValue( 'risk_cover_threshold', 1000 );
                                 if ( $riskCoverEnabled && $overThreshold ) {
-                                    $riskCover = 1;
+                                    $riskCover = true;
                                 }
 
                                 $data = [
-                                    'to_town_id' => $package['destination']['to_town_id'],
-                                    'from_town_id' => $package['destination']['from_town_id'],
-                                    'to_location_type' => $package['destination']['to_location_type'],
-                                    'from_location_type' => $package['destination']['from_location_type'],
-                                    'cover' => $riskCover,
-                                    'weight' => $package['cart']['weight'],
-                                    'num_package' => $package['cart']['count'],
+                                    'delivery_town' => $package['destination']['to_town_id'],
+                                    'collection_town' => $package['destination']['from_town_id'],
+                                    'delivery_location_type' => $package['destination']['to_location_type'],
+                                    'collection_location_type' => $package['destination']['from_location_type'],
+                                    'risk_cover' => $riskCover,
+                                    //'weight' => $package['cart']['weight'],
```

Comment:
Careful about these commented out fields, either it was left commented out because you were testing or it should be removed.

Reply:
I forgot to remove that originally.
It's been removed.